src/write_gcm.c

```diff
@@ -3019,25 +3020,16 @@ static int wg_json_CreateTimeSeries(
                 "type_instance: %s metric type must be string.",
                 head->key.plugin, head->key.plugin_instance, head->key.type,
                 head->key.type_instance);
-          continue;
-        }
-        const char *pref = custom_metric_prefix;
-        if (strncmp(entry->value.value_text, pref, strlen(pref)) != 0) {
-          ERROR("write_gcm: plugin: %s, plugin_type: %s, metric_type: %s, "
-                "type_instance: %s metric type %s is not a custom metric "
-                "(should start with '%s').",
-                head->key.plugin, head->key.plugin_instance, head->key.type,
-                head->key.type_instance, entry->value.value_text, pref);
-          continue;
+          continue next_payload;
         }
       }
       const char *key_pref = custom_metric_label_prefix;
       if (strncmp(entry->key, key_pref, strlen(key_pref)) == 0) {
         if (entry->value.value_type != wg_typed_value_string) {
-          ERROR("write_gcm: plugin: %s, plugin_type: %s, metric_type: %s, "
-                "type_instance: %s metric label %s is not a string.",
-                head->key.plugin, head->key.plugin_instance, head->key.type,
-                head->key.type_instance, entry->key);
+          WARNING("write_gcm: plugin: %s, plugin_type: %s, metric_type: %s, "
+                  "type_instance: %s metric label %s is not a string.",
```

Comment:
The API will actually create metric descriptors with string labels, and then attempt to write the time series with this label into that metric descriptor, which will fail. The error returned from the API will be something like "label <foo> is not a string". In combination with this message, it should direct the users to the source of the problem.